pyspecdata/core.py

```diff
@@ -130,7 +130,14 @@ def apply_oom(average_oom,numbers,prev_label=''):
             raise ValueError(strm("you passed",average_oom,"which I can't find a prefix for"))
     else:
         oom_index = nonzero(eq)[0][0]
-    numbers[:] /= 10.**oom_values[oom_index]
+    print('type of numbers',numbers.dtype)
+    orig_dtype = numbers.dtype
+    oom = oom_values[oom_index]
+    print('type of oom', oom.dtype)
+    numbers = numbers.astype(float)
```

Comment:
so, this actually won't work, b/c it creates a copy of numbers, and the point of the function seems to be to modify the `numbers` array in-place

Reply:
(if you tried out this function by itself, you would see that it returns the prefix, but the numbers aren't scaled -- this will create a problem with the units)